### PR TITLE
Add Id token validation fix and HTTP Session Wrapper for the OIDC Manger.

### LINKED ISSUE #4: ID Token Validation in OIDC flows.
**Description:**
In the oidc-sdk, its need to validate the ID Token as per the openid-connect spec [1]. Listing the points below for the ease of reference.

1. If the ID Token is encrypted, decrypt it using the keys and algorithms that the Client specified during Registration that the OP was to use to encrypt the ID Token. If encryption was negotiated with the OP at Registration time and the ID Token is not encrypted, the RP SHOULD reject it.
2. The Issuer Identifier for the OpenID Provider (which is typically obtained during Discovery) MUST exactly match the value of the iss (issuer) Claim.
3. The Client MUST validate that the aud (audience) Claim contains its client_id value registered at the Issuer identified by the iss (issuer) Claim as an audience. The aud (audience) Claim MAY contain an array with more than one element. The ID Token MUST be rejected if the ID Token does not list the Client as a valid audience, or if it contains additional audiences not trusted by the Client.
4. If the ID Token contains multiple audiences, the Client SHOULD verify that an azp Claim is present.
5. If an azp (authorized party) Claim is present, the Client SHOULD verify that its client_id is the Claim Value.
6. If the ID Token is received via direct communication between the Client and the Token Endpoint (which it is in this flow), the TLS server validation MAY be used to validate the issuer in place of checking the token signature. The Client MUST validate the signature of all other ID Tokens according to JWS [JWS] using the algorithm specified in the JWT alg Header Parameter. The Client MUST use the keys provided by the Issuer.
7. The alg value SHOULD be the default of RS256 or the algorithm sent by the Client in the id_token_signed_response_alg parameter during Registration.
8. If the JWT alg Header Parameter uses a MAC based algorithm such as HS256, HS384, or HS512, the octets of the UTF-8 representation of the client_secret corresponding to the client_id contained in the aud (audience) Claim are used as the key to validate the signature. For MAC based algorithms, the behavior is unspecified if the aud is multi-valued or if an azp value is present that is different than the aud value.
9. The current time MUST be before the time represented by the exp Claim.
10. The iat Claim can be used to reject tokens that were issued too far away from the current time, limiting the amount of time that nonces need to be stored to prevent attacks. The acceptable range is Client specific.
11. If a nonce value was sent in the Authentication Request, a nonce Claim MUST be present and its value checked to verify that it is the same value as the one that was sent in the Authentication Request. The Client SHOULD check the nonce value for replay attacks. The precise method for detecting replay attacks is Client specific.
12. If the acr Claim was requested, the Client SHOULD check that the asserted Claim Value is appropriate. The meaning and processing of acr Claim Values is out of scope for this specification.
13. If the auth_time Claim was requested, either through a specific request for this Claim or by using the max_age parameter, the Client SHOULD check the auth_time Claim value and request re-authentication if it determines too much time has elapsed since the last End-User authentication.

Need to test whether each of above points are covered by the SDK flow.

[1] - https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation

### PR DESCRIPTION
**Pull request by ChamathNS**
## Purpose
- Implement ID Token validation.
- Add support for a HTTP session based wrapper for the DefaultOIDCManager class.

## Related PRs
resolves #4

### CODE REVIEW

File: io.asgardio.java.oidc.sdk/src/test/java/io/asgardio/java/oidc/sdk/HTTPSessionBasedOIDCProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+        verify(session).setAttribute(SSOAgentConstants.REQUEST_CONTEXT, requestContext);
+    }
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+}
```
Comment (gayashanbc) [inline io.asgardio.java.oidc.sdk/src/test/java/io/asgardio/java/oidc/sdk/HTTPSessionBasedOIDCProcessorTest.java:141]: Add a newline at the end of the file

File: io.asgardio.java.oidc.sdk/src/main/java/io/asgardio/java/oidc/sdk/request/OIDCRequestBuilder.java [hunk 2]
```diff
@@ -58,62 +63,83 @@ public OIDCRequestBuilder(OIDCAgentConfig oidcAgentConfig) {
...
+        URI logoutRequestURI;
+
+        try {
+            logoutRequestURI = new LogoutRequest(logoutEP, jwtIdToken, redirectionURI, state).toURI();
+        } catch (Exception e) {
+            throw new SSOAgentServerException(e.getMessage(), e);
         }
-        return new LogoutRequest(logoutEP, jwtIdToken, redirectionURI, stateParam).toURI().toString();
+
+        return new io.asgardio.java.oidc.sdk.request.model.LogoutRequest(logoutRequestURI, requestContext);
```
Comment (gayashanbc) [inline io.asgardio.java.oidc.sdk/src/main/java/io/asgardio/java/oidc/sdk/request/OIDCRequestBuilder.java:137]: Can we import the class without prepending the package name everywhere?

File: io.asgardio.java.oidc.sdk/src/test/java/io/asgardio/java/oidc/sdk/validators/IDTokenValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+        IDTokenClaimsSet claimsSet = validator.validate(nonce);
+        assertEquals(claimsSet.getNonce(), nonce);
+    }
+
+    @AfterMethod
+    public void tearDown() {
+
+        closeJadler();
+    }
+}
```
Comment (gayashanbc) [inline io.asgardio.java.oidc.sdk/src/test/java/io/asgardio/java/oidc/sdk/validators/IDTokenValidatorTest.java:292]: Add a newline at the end

File: io.asgardio.java.oidc.sdk/src/main/java/io/asgardio/java/oidc/sdk/request/OIDCRequestBuilder.java (hunk 2 above)
Comment (ChamathNS) [reply io.asgardio.java.oidc.sdk/src/main/java/io/asgardio/java/oidc/sdk/request/OIDCRequestBuilder.java:137]: We already have a import called com.nimbusds.openid.connect.sdk.LogoutRequest which refers to the LogoutRequest class in the nimbusds package. 
To distinguish between the two LogoutRequest classes, the package name had to be prepended.
Suggestions to resolve this are welcome.

File: io.asgardio.java.oidc.sdk/src/test/java/io/asgardio/java/oidc/sdk/validators/IDTokenValidatorTest.java (hunk 3 above)
Comment (ChamathNS) [reply io.asgardio.java.oidc.sdk/src/test/java/io/asgardio/java/oidc/sdk/validators/IDTokenValidatorTest.java]: Resolved with https://github.com/asgardio/asgardio-java-oidc-sdk/pull/12/commits/8935b73e06c06ac1e1f216c89c0b19e175e1c196.

File: io.asgardio.java.oidc.sdk/src/test/java/io/asgardio/java/oidc/sdk/HTTPSessionBasedOIDCProcessorTest.java (hunk 1 above)
Comment (ChamathNS) [reply io.asgardio.java.oidc.sdk/src/test/java/io/asgardio/java/oidc/sdk/HTTPSessionBasedOIDCProcessorTest.java]: Resolved with https://github.com/asgardio/asgardio-java-oidc-sdk/pull/12/commits/8935b73e06c06ac1e1f216c89c0b19e175e1c196.

**Review by gayashanbc — APPROVED** [review]

**Review by darshanasbg — APPROVED** [review]

**Pull request merged**